docs/guides/protect-page-login/react.mdx

```diff
@@ -0,0 +1,108 @@
+---
+id: react
+title: Protect a Page with Login in React
+sidebar_label: React
+---
+
+```mdx-code-block
+import Teaser from '../_common/teaser.mdx'
+import mp4 from '../_static/react/screencast.mp4'
+import webm from '../_static/react/screencast.webm'
+
+<Teaser
+  framework={<a href="https://reactjs.org/">React</a>}
+  mp4={mp4}
+  webm={webm}
+/>
+```
+
+## Create React App
+
+First we create a new React project:
+
+```shell-session
+npx create-react-app --typescript my_project
+cd my_project
+```
+
+## Install Ory SDK
+
+Use `npm` to install the Ory SDK:
+
+```shell-session
+npm i --save @ory/client
+```
+
+## Require Login to Access the App Page
+
+Next we add a session check to the default home page of the React example application.The highlighted code is what we added to
+check whether the user is signed in, get the logout url flow and redirect them to the login page if not. There is also a welcome message with the registered user email and a logout link:
+
+```mdx-code-block
+import CodeBlock from '@theme/CodeBlock'
+import App from '!!raw-loader!../../../code-examples/protect-page-login/react/src/App.tsx'
+
+<CodeBlock language="tsx">{App}</CodeBlock>
+```
+
+## Run Ory APIs on `localhost`
+
+The Ory security model uses HTTP cookies to manage sessions, tokens, and cookies. To do that, Ory APIs must be exposed on the same
+domain as your application.
+
+In the case of this example, the domain of your application, which runs on your local machine, is `localhost`.
+
+:::info
+
+When developing locally, use either `localhost` or `127.0.0.1`, but not both. Although technically these mean the same thing,
+they're different hostnames.
+
+Using both interchangeably in your code can cause problems with the origin of HTTP cookies sent by the application. When that happens
+Ory APIs might not be able to read the cookies sent by your app.
+
+:::
+
+To get your application that runs locally and Ory APIs on the same domain, use Ory Tunnel - a development tool bundled with Ory CLI.
+It's like a microservice - an authentication API server on your domain!
+
+```shell-session
+npx @ory/cli tunnel --dev http://localhost:3000
+```
+
+Ory APIs are now mirrored on `http://localhost:4000`. Use that URL as the `baseUrl` for the `@ory/client` SDK (see code above).
+The `--dev` flag disables security checks for easier integration and should not be used when deploying the Tunnel
+to a staging environment.
+
+Then open [http://localhost:3000](http://localhost:3000) in your browser. are presented with Ory's Sign In page! Let's click on
+sign up and create your first user!
```

Comment:
one last nit

this section probably needs to be under the `Run your React App` heading after the `npm run start` command?

Reply:
Indeed! Also pushing a change for this part `in your browser. are presented with Ory's Sign In page! Let's click on` I think it makes more sense after the dot: `After loading you now are presented...` Already pushed the change (need to patch nextjs page too, have same exact phrase). What do you think?

Reply:
Perfect then!